support/metrics/metricsTracker.go

```diff
@@ -286,3 +350,36 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+func mergeEventProps(commonProps, eventProps map[string]interface{}) (map[string]interface{}, error) {
+	m := make(map[string]interface{})
+	for k, v := range commonProps {
+		m[k] = v
+	}
+
+	for k, v := range eventProps {
+		m[k] = v
+	}
+
+	return m, nil
+}
+
+func toMapStringInterface(v interface{}) (map[string]interface{}, error) {
+	b, e := json.Marshal(v)
+	if e != nil {
+		return nil, fmt.Errorf("could not marshal interface to json: %s", e)
+	}
+
+	var i interface{}
```

Comment:
wondering if this can be a `map[string]interface{}`?

then we won't need the cast below

Reply:
I think it can be, but it's recommended everywhere I could find resources to unmarshal arbitrary structs to `map[string]interface{}`. See a relevant Golang [blogpost](https://blog.golang.org/json), under "Decoding arbitrary data".

With that said, I can't find a functional reason for this and it saves a cast that should get caught in the Unmarshal, so making this change.